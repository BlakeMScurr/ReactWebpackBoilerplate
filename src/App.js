import React from "react";
import logo from "./logo.svg";
import "./App.less";
import { observer } from "mobx-react";
import { observable } from "mobx";

@observer
class App extends React.Component {
    @observable tree = "sex";
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <button
                        id={"test"}
                        onClick={() => {
                            this.tree = "dream" + Date.now();
                        }}
                    >
                        DCCC
                    </button>
                    <p>
                        {this.tree} Edit <code>src/App.js</code> and save to reload.asdsadasdsd asd
                    </p>
                    <a
                        className="App-link"
                        href="https://reactjs.org"
                        target="_blank"
                        rel="noopener noreferrer"
                    >
                        Learn React
                    </a>
                </header>
            </div>
        );
    }
}

export default App;
