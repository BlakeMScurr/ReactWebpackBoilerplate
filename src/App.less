.chat {
    width: 960px;
}
