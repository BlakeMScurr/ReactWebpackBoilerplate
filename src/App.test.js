import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
import { render, fireEvent, cleanup } from "react-testing-library";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

test("A[[p222 changes the text after click", () => {
    // Render a checkbox with label in the document
    const AppRend = shallow(<App />);
    expect(AppRend.find("#test").text()).toEqual("DCCC");
    // expect(checkbox.text()).toEqual("On");
});

test("Link changes the class when hovered", () => {
    const component = renderer.create(<App />);

    let tree = component.toJSON();
    expect(tree).
    toMatchSnapshot();
    //
    // // manually trigger the callback
    // tree.props.onMouseEnter();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    //
    // // manually trigger the callback
    // tree.props.onMouseLeave();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
});
