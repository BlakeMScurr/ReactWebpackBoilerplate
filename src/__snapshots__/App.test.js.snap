// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      id="test"
      onClick={[Function]}
    >
      DCCCss
    </button>
    <p>
      sex
       Edit 
      <code>
        src/App.js
      </code>
       and save to reload.asdsadasdsd asd
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
